### ISSUE TITLE
Unable to save refresh tokens; Changes to the project config are not possible while in read-only

### ISSUE THREAD

**Issue by nateiler** [flipboxfactory/patron#7]
When admin change are prohibited and a refresh token flow occurs, an error is thrown: **Changes to the project config are not possible while in read-only**

**Comment by dustinwalker** [comment]
Following up from email here: We've been seeing this on a few versions, including the version we are currently on Craft 3.5.12.1. Unfortunately, I do not recall for certain if we saw this on 3.4.x. 

Let me know if you need any other details, always happy to send files by email if helpful.

Thanks!
Dustin

**Comment by dustinwalker** [comment]
@nateiler Hey Nate, is there anything I could modify on my end to prevent this or should I plan on disabling project config for launch? It's not clear to me exactly when this is triggered. If it's only once after deployment that would be fine, but don't want to subject the client's customers to that message.

**Comment by nateiler** [comment]
Thanks for the nudge.

I was able to quickly identify the issue which was causing the error to be thrown; but upon investigating further I realized some other aspects of the project config needed to be addressed.  

I have these up on `develop` branch now doing some testing.  Feel free to take a look and test if you would like.

Also, I'm going to suggest rebuilding the project config `php craft project-config/rebuild` to ensure the correct configuration is saved for the plugin.

**Comment by dustinwalker** [comment]
Thanks, I was out for a few days but going to test this out now. I will be sure to rebuild the config.

**Comment by dustinwalker** [comment]
@nateiler I haven't had the error thrown in testing using the develop branch. I'm not sure the scenarios that triggered it before, so I'll test it again tomorrow AM.

Would this be safe to apply to live if that checks out? We launched this site late last week and I'd love to tick this one off on our to-do list. 

Thanks again!

**Comment by nateiler** [comment]
I released version 2.3.0 (https://github.com/flipboxfactory/patron/releases/tag/2.3.0).  

Let me know if you experience anything else.

**Comment by dustinwalker** [comment]
Thanks Nate. I'll open a new issue if we run into anything but working great so far.
